modal/_output.py

```diff
@@ -448,15 +449,17 @@ async def _get_logs():
                 if pty_shell_finish_event:
                     print("ERROR: concurrent PTY shells are not supported.")
                 else:
+                    output_mgr.flush_lines()
                     output_mgr.hide_status_spinner()
                     output_mgr._visible_progress = False
                     pty_shell_finish_event = asyncio.Event()
+                    pty_shell_task_id = log_batch.task_id
                     asyncio.create_task(stream_pty_shell_input(client, log_batch.pty_exec_id, pty_shell_finish_event))
             else:
                 for log in log_batch.items:
                     await _put_log(log_batch, log)
 
-            if log_batch.eof:
+            if log_batch.eof and log_batch.task_id == pty_shell_task_id:
                 await stop_pty_shell()
```

Comment:
I'm trying to detect when the current PTY shell's task ends, and stop the pty shell. However, this doesn't seem to work very reliably.